### ISSUE TITLE
Include a more deliberate integration of accessibility

### ISSUE THREAD

**Issue by Steve-Noble** [w3c/mathml#282]
### MathML and Accessibility
I propose we consider a more deliberate approach to connecting accessibility with MathML usage in the new specification. In the current spec, there are only a small number of passing references to accessibility. Mathematics itself is a language intended for more than just visual rendering, inspection and manipulation, and indeed the current language we have in the spec alludes to communication the "meaning" of mathematics with language about semantics and intent, which have a direct connection to mathematics accessibility. Some other markup specs explicitly point out accessibility considerations, so I suggest we look at the methods employed in some of these and decide how we want to approach it. The three general techniques I see used include:

- an embedded template approach used throughout the spec
- including a separate section in the spec
- placing accessibility documentation in an appendix

### Template approach
Examples HTML https://html.spec.whatwg.org/ 
In this spec, the authors have implemented a template where every element tag includes a description as below:

- Accessibility considerations:
  - For authors
  - For implementers

If you follow these links, you will notice that they lead to and extensive table which makes explicit connections to ARIA, including when there is no corresponding ARIA role. In cases where special accessibility issues come into play, a “note” is added (for example 4.8.9, the “video” element; 4.8.10, the “audio” element), or these concerns are noted at the end of the element discussion (for example, 4.9.1, the “table” element), or throughout the element discussion (for example, numerous subheadings under 4.12 the “canvas” element; numerous subheadings under 4.13 “Custom elements”)

Another example is Map Markup Language https://maps4html.org/MapML/spec/ 
“This specification describes Map Markup Language (MapML), which is an extended subset of HTML, for maps. MapML extends the semantics of several HTML Standard elements, and specifies a small set of new, mapping-specific elements, in the HTML namespace." Since this spec is considered an “extended subset of HTML” a similar template approach has been used.

However, it is unlikely that every MathML element will have a specific accessibility consideration, and individually connecting each element to an ARIA role would probably not be useful or appropriate for most MathML elements. Therefore, it probably makes more sense to describe accessibility considerations only within the particular elements where we can identify that something needs to be stated, and the template approach seems unneeded.

### Including an Accessibility chapter
Example: EPUB 3.3 https://www.w3.org/TR/epub-33/ 
The word “accessibility” can be found 42 times in the spec, and various accessibility considerations get mentioned as specific elements are described. However, the use the model of labeling a separate Accessibility section (Section 8), but then they point readers to another document: the EPUB Accessibility spec https://www.w3.org/TR/epub-a11y-11/ 

They defend this approach by stating that by separating these two specifications, the EPUB Accessibility Specification can be developed on its own timeline and the whole spec won’t have to be updated in order to address needed changes in the Accessibility Specification. While this may make a lot of sense, we should also consider that fact that implementers who are trying to support the primary technology specification can easily consider the Accessibility Specification as something they don’t need to be concerned with, or that they simply overlook it because it isn’t in the “real” specification.

### Putting the Accessibility discussion in an appendix
SVG put accessibility documentation into the appendix, but also interweaves important accessibility considerations which impact particular elements as these come up in the spec. You can spot these by pulling up the full spec (https://svgwg.org/svg2-draft/single-page.html) and searching for “accessib” 

So, that is not too different than the EPUB spec, other than the fact that putting the "accessibility" section in an appendix has the connotation of making it unimportant.

**Comment by davidcarlisle** [comment]
@Steve-Noble  I added a stub file at

html view: https://w3c.github.io/mathml/#mathml-accessibility

source: https://github.com/w3c/mathml/blob/gh-pages/src/accessibility.html

It is currently an appendix but it can easily be made a chapter if we decide that later by moving the location of the inclusion in index.html, the markup is the same either way.

the only content is a link to this issue.

**Comment by davidcarlisle** [comment]
Most of the Authoring guidelines for accessibility will necessarily be non normative so it may turn out that more of the accessibiliy text ends up in the Notes on MathML at


https://w3c.github.io/mathml-docs/notes-on-mathml/#notes-on-mathml-accessibility

That is currently empty but I think we could get an initial draft by taking the full text of @NSoiffer and @samdooley document "Math Accessibility Primer"  which was produced under the mathml-refresh banner

https://mathml-refresh.github.io/discussion-papers/accessibility

That is already in markdown so could easily be moved, if the authors think that's a good plan?

we could also steal bits of

https://mathml-refresh.github.io/discussion-papers/a11y-needs

but I think the primer is perhaps closest to the target for Notes-On-MathML.

**Comment by Steve-Noble** [comment]
I would be fine with using that as a starting place. I just want to be sure that accessibility doesn't become an afterthought which nobody looks at because it is not prominently called out in the formal spec. I suggest the [approach used in the EPUB spec](https://www.w3.org/TR/epub-33/), where Accessibility is given a chapter designation as [Chapter 10](https://www.w3.org/TR/epub-33/#sec-accessibility), but here there is only a few paragraphs of text and the reader is pointed to the [EPUB Accessibility spec](https://www.w3.org/TR/epub-a11y-11/#). To me, this would give accessibility a greater prominence in the MathML 4 spec, while not making the spec appreciable longer.

**Comment by davidcarlisle** [comment]
@Steve-Noble Thanks, so I just pushed it it the notes document

https://w3c.github.io/mathml-docs/notes-on-mathml/#notes-on-mathml-accessibility

The text will revising but it sems a better basis than starting with an empty file.

The EPUB approach of p.utting most/all of the accessibility requirements in the non normaive document and just having a small section in the main spec sounds like a good plan to me. There may in the end be some normative things we can say (in relation to what does and does not get put in the accesibility tree), but we might not be in a position to standardise that in the timeframe of this iteration of the spec, so I think most of the text will be non normative authoring guidelines, which fits with the model used by epub where almost every section of  epub-a11y-11 is non normative.

**Comment by clapierre** [comment]
I am wondering about two things in terms of accessibility the first is fallback which I would think is the alt-text for the math equation, and the second is the growing popularity of "Read Now" functionality of EPUB reading systems and what is spoken when a reading system encounters a mathML equation when in this mode of automatic reading aloud (which is separate than what a Screen Reader would speak / interact with mathML).  At the very least this should be spelled out in the Normative specification on what would be spoken.

**Comment by davidcarlisle** [comment]
@clapierre  currently the mathml full alttext attribute is not supported in Core, although (like intent) we could add it to the schema so it could be used in valid documents and used on systems not using the mathml core rendering. (I must admit I'm not sure I fully understand the implications of Read Now as opposed to a Screen reader.)

**Comment by NSoiffer** [comment]
Just to second @Steve-Noble's suggestion, SVG's editor's draft has [an informative appendix ](https://svgwg.org/svg2-draft/access.html) whose first section takes some WCAG2 guidelines and says how to handle them in SVG. That might be a good thing for the MathML spec (hint, hint, to Steve...). A second section points to other specs such as the accessibility platform mappings.

Note: SVG's official spec's are over a decade old so not representative of current W3C requirements.

**Comment by NSoiffer** [comment]
My understanding of how the "Read Now" functionality differs from a screen reader is that it is "built in" to the EPUB app. As such, it tends to be less powerful, less configurable, and not as personal -- some other app like Adobe Acrobat might have a "Read Aloud" feature which accomplishes the same thing but uses different key strokes, etc., and has a different look when doing the reading.  [Here's a video of the thorium EPUB reader showing readaloud](https://youtu.be/3FFkrd73lEI).

Having said that, there's no technical reason why an EPUB application can't do the sync highlighting of the math just as it does for the text if it took advantage of the MathML rather than just using `alttext`. Both Volker's SRE and MathCAT support sync highlighting so they don't need to start from scratch to make it work for MathML.

**Comment by Steve-Noble** [comment]
I should mention that Pearson and other large assessment vendors make heavy use of the math alttext attribute. This is done because in high stakes assessments, state education agencies want to have very precise control over how the math is voiced in TTS engines. Sounds like I need to make a case for the support of the alttext attribute in Core, which probably belongs as a separate issue.

**Comment by NSoiffer** [comment]
Done.
